AWSTranscribeStreaming/Internal/AWSSRWebSocketDelegateAdaptor.h

```diff
@@ -16,13 +16,14 @@
 #import <Foundation/Foundation.h>
 #import "AWSSRWebSocket.h"
 #import "AWSTranscribeStreamingClientDelegate.h"
+#import "AWSTranscribeStreamingWebSocketProviderDelegate.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 /**
  An AWSSRWebSocketDelegate adaptor for AWSTranscribeStreamingClientDelegate
  */
-@interface AWSSRWebSocketDelegateAdaptor : NSObject <AWSSRWebSocketDelegate>
+@interface AWSSRWebSocketDelegateAdaptor : NSObject <AWSTranscribeStreamingWebSocketProviderDelegate, AWSSRWebSocketDelegate>
```

Comment:
this conforming to both protocols is my fundamental problem that makes a really ugly solution. any tips here would be much appreciated.

Reply:
Agree that it's sort of 'ugly'.. hmmm.  Correct me if i'm wrong but, once we deprecate socket rocket, two things will happen:
1.  This class will eventually go away
2.  We will create a new class which conforms to AWSTranscribeStreamingWebSocketProviderDelegate which is responsible for making callbacks to the clientDelegate(AWSTranscribeStreamingClientDelegate)

If this is the case, i'm not super concerned about this ugliness.

Reply:
not sure we are deprecating Socket Rocket tho? I thought the whole point was to keep it so existing customers can use it as a default? and yes you're one hundred percent right if it will be deprecated

Reply:
deferring this one to @palpatim when he returns

Reply:
AFAIK, there's no plan to deprecate SocketRocket. The only alternative Web Socket solutions are either pure Swift (StarScream) or iOS 13+ (native NSURLSession WebSocket tasks).

Reply:
@palpatim thanks for confirming. do you have any suggestions here for the ugliness of the 2 protocol situation here?

Reply:
resolving this as per discussion with @palpatim of it not being code smell